handlers/jaeger.go

```diff
@@ -3,42 +3,44 @@ package handlers
 import (
 	"fmt"
 	"net/http"
-	"net/http/httputil"
-	"net/url"
-	"strings"
 
 	"github.com/kiali/kiali/config"
 	"github.com/kiali/kiali/log"
 	"github.com/kiali/kiali/services/models"
 )
 
-func ProxyJaeger(w http.ResponseWriter, r *http.Request) {
-	jaegerConfig := config.Get().ExternalServices.Jaeger
-	// It assumes that jaeger internally is accessible through http. This is how it works in Istio 1.0 GA.
-	url, err := url.Parse(fmt.Sprintf("http://%s.%s:%s/", jaegerConfig.Service, jaegerConfig.ServiceNamespace, jaegerConfig.ServicePort))
-	if err != nil {
-		log.Error(err)
-		return
-	}
-	proxy := httputil.ReverseProxy{Director: func(r *http.Request) {
-		r.URL.Scheme = url.Scheme
-		r.URL.Host = url.Host
-		r.URL.Path = strings.Replace(r.URL.Path, "/api/jaeger", "", -1)
-		r.Host = url.Host
-	}}
-	proxy.ServeHTTP(w, r)
-}
-
-// GetJaegerInfo provides the proxy Jaeger URL
+// Get JaegerInfo provides the proxy Jaeger URL
 func GetJaegerInfo(w http.ResponseWriter, r *http.Request) {
 	scheme := "http"
 	if r.URL.Scheme != "" {
 		scheme = r.URL.Scheme
 	}
-	// r.Host can be set as "host" or "host:port", we always remove the :port as it is updated by the proxy
-	kialiHost := strings.Split(r.Host, ":")
+	jaegerConfig := config.Get().ExternalServices.Jaeger
+
+	spec, err := getService(jaegerConfig.ServiceNamespace, jaegerConfig.Service)
```

Comment:
I am not sure if we should fully delegate this into the configuration.
I mean, Istio by default does not expose Jaeger, so it is clear that we should ask a step to the user to publish the URL.
As community is using kubernetes on several vendors, perhaps it's just a matter to indicate this step instead to try to resolve it automatically.
I do not know, I guess this happens same as graphana links, as it is same scenario.

Reply:
We are doing the same for the grafana info. So we should change for grafana too.